gcloud/pubsub/api.py

```diff
@@ -52,3 +52,49 @@ def list_topics(page_size=None, page_token=None,
 
     path = '/projects/%s/topics' % project
     return connection.api_request(method='GET', path=path, query_params=params)
+
+
+def list_subscriptions(page_size=None, page_token=None, topic_name=None,
+                       project=None, connection=None):
+    """List subscriptions for a given project.
+
+    :type page_size: int
+    :param page_size: maximum number of topics to return, If not passed,
+                      defaults to a value set by the API.
+
+    :type page_token: string
+    :param page_token: opaque marker for the next "page" of topics. If not
+                       passed, the API will return the first page of topics.
+
+    :type topic_name: string
+    :param topic_name: limit results to subscriptions bound to the given topic.
+
+    :type project: string
+    :param project: project ID to query.  If not passed, defaults to the
+                    project ID inferred from the environment.
+
+    :type connection: :class:`gcloud.pubsub.connection.Connection`
+    :param connection: connection to use for the query.  If not passed,
+                       defaults to the connection inferred from the
+                       environment.
+
+    :rtype: dict
+    :returns: keys include ``subscriptions`` (a list of subscription mappings)
+              and ``nextPageToken`` (a string:  if non-empty, indicates that
+              more topics can be retrieved with another call (pass that
+              value as ``page_token``).
+    """
+    params = {}
+
+    if page_size is not None:
+        params['pageSize'] = page_size
+
+    if page_token is not None:
+        params['pageToken'] = page_token
+
+    if topic_name is None:
+        path = '/projects/%s/subscriptions' % project
```

Comment:
`project` will never be None once we are done folding in the "implicit environment" stuff  we will update in a subsequent PR to infer the "default" project from the environment, if it is not passed explicitly to the topic:  at that point, if the inference fails, we will raise an exception from `Topic.__init__`.